### PR TITLE
W-12288635: Remove SpanInitialInfo Providers

### PR DESCRIPTION
**Pull request by fsgonz**

### CODE REVIEW

File: core/src/main/java/org/mule/runtime/core/api/tracing/customization/ComponentExecutionInitialSpanInfo.java [hunk 1]
```diff
@@ -28,55 +26,37 @@ public class ComponentExecutionInitialSpanInfo implements InitialSpanInfo {
 
 
   public static final String LOCATION_KEY = "location";
-  public static final String CORRELATION_ID_KEY = "correlation.id";
-  public static final String THREAD_START_ID_KEY = "thread.start.id";
-  public static final String THREAD_START_NAME_KEY = "thread.start.name";
   // These are location, correlation.id, thread.start.id, thread.start.name
```
Comment (sofiamorseletto) [inline core/src/main/java/org/mule/runtime/core/api/tracing/customization/ComponentExecutionInitialSpanInfo.java:29]: Update comment with the one attribute that is counted now

File: core/src/main/java/org/mule/runtime/core/privileged/processor/chain/AbstractMessageProcessorChain.java [hunk 2]
```diff
@@ -494,15 +486,37 @@ private List<ReactiveInterceptor> resolveInterceptors() {
...
+          .doOnNext(event -> beforeComponentProcessingStrategy((Processor) processor, processorPath, processorInitialSpanInfo,
+                                                               assertion, event))
           .transform(next)
           .map(result -> afterComponentProcessingStrategy((Processor) processor, processorPath, result));
     });
 
     return interceptors;
   }
 
+  private InitialSpanInfo getProcessorIntiialSpanInfo(ReactiveProcessor processor) {
```
Comment (sofiamorseletto) [inline core/src/main/java/org/mule/runtime/core/privileged/processor/chain/AbstractMessageProcessorChain.java:504]: getProcessorInitialSpanInfo*

File: core/src/main/java/org/mule/runtime/core/privileged/processor/chain/AbstractMessageProcessorChain.java [hunk 3]
```diff
@@ -494,15 +486,37 @@ private List<ReactiveInterceptor> resolveInterceptors() {
...
+    if (processor instanceof Component) {
+      // If this is a component we create the span with the corresponding name.
+      initialSpanInfo =
+          new ComponentExecutionInitialSpanInfo((Component) processor);
+    } else {
+      // Other processors are not exported
+      initialSpanInfo = new NoExportFixedNameInitialSpanInfo(UNKNOWN);
+    }
+
+    InitialSpanInfo processorInitialSpanInfo = initialSpanInfo;
```
Comment (sofiamorseletto) [inline core/src/main/java/org/mule/runtime/core/privileged/processor/chain/AbstractMessageProcessorChain.java:516]: Is this line necessary?

File: core/src/main/java/org/mule/runtime/core/privileged/processor/chain/AbstractMessageProcessorChain.java [hunk 4]
```diff
@@ -527,11 +541,14 @@ private CoreEvent afterComponentProcessingStrategy(Processor processor, String p
     }
   }
 
-  private void beforeComponentProcessingStrategy(Processor processor, String processorPath, CoreEvent event) {
+  private void beforeComponentProcessingStrategy(Processor processor, String processorPath, InitialSpanInfo initialSpanInfo,
+                                                 Assertion assertion, CoreEvent event) {
     // The span corresponding to the processor has to be created here because if the processor
     // cannot process a message (by the canProcessMessage condition below), the exception will be considered
     // parte of the execution of the processor.
```
Comment (sofiamorseletto) [inline core/src/main/java/org/mule/runtime/core/privileged/processor/chain/AbstractMessageProcessorChain.java:548]: part*

**Review by sofiamorseletto — CHANGES_REQUESTED** [review]

**Review by sofiamorseletto — APPROVED** [review]

**Review by IvanAndresFritzler — APPROVED** [review]

**Comment by fsgonz** [comment]
--validate

**Pull request merged**